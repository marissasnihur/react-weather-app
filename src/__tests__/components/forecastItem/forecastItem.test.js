import React from 'react';
import ReactDOM from 'react-dom';

import ForecastItem from '../../../components/ForecastItem/ForecastItem';

describe('ForecastItem component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ForecastItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render with props correctly', () => {
    const div = document.createElement('div');
    const forecastItem = (
      <ForecastItem icon="clear" high="30" low="-30" dayOfWeek="Fri"/>
    );
    ReactDOM.render(forecastItem, div);
    expect(div.firstChild.childNodes[0].getAttribute('class')).toBe('forecastIcon wi clear');
    expect(div.firstChild.childNodes[1].textContent).toBe('30');
    expect(div.firstChild.childNodes[2].textContent).toBe('-30');
    expect(div.firstChild.childNodes[3].textContent).toBe('Fri');
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    const forecastItem = (
      <ForecastItem icon="clear" high="30" low="-30" dayOfWeek="Fri"/>
    );
    ReactDOM.render(forecastItem, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});
