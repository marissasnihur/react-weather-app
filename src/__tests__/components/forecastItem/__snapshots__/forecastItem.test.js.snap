// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem component should match snapshot 1`] = `
<div>
  <li
    class="forecastItem"
  >
    <span
      class="forecastIcon wi clear"
    />
    <div
      class="cardTemperature"
    >
      30
    </div>
    <div
      class="cardTemperature"
    >
      -30
    </div>
    <p
      class="forecastDay"
    >
      Fri
    </p>
  </li>
</div>
`;
